### PR TITLE
chore: Simplify cert-manager certs and support existingIssuer

### PR DESCRIPTION
**Pull request by mkilchhofer**
Improve 2 things related to cert-manager certificates:
- Simplify the cert-manager resources. 2 of 4 resources are not required and I never seen such setups in the community
- add support for an existing Issuer / ClusterIssuer.

Somehow related to #371 as this introduced the usage of cert-manager.

### Checklist

- [ ] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)*
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*

Fixes #

### CODE REVIEW

File: keda/templates/metrics-server/apiservice.yaml [hunk 1]
```diff
@@ -3,11 +3,7 @@ kind: APIService
 metadata:
   annotations:
     {{- if .Values.certificates.certManager.enabled }}
-    {{- if .Values.certificates.certManager.generateCA }}
-    cert-manager.io/inject-ca-from: {{ .Release.Namespace }}/{{ .Values.operator.name }}-ca
-    {{- else }}
-    cert-manager.io/inject-ca-from-secret: {{ .Release.Namespace }}/{{ .Values.certificates.certManager.caSecretName }}
-    {{- end }}
+    cert-manager.io/inject-ca-from: {{ .Release.Namespace }}/{{ .Values.operator.name }}-tls-certificates
```
Comment (mkilchhofer) [inline keda/templates/metrics-server/apiservice.yaml:6]: was configured like its documented inside the cert-manager docs.
I'd suggest to follow 1:1 the docs of cert-manager:
https://cert-manager.io/docs/concepts/ca-injector/#injecting-ca-data-from-a-certificate-resource

Comment (JorTurFer) [reply keda/templates/metrics-server/apiservice.yaml:6]: What do you mean?

Comment (mkilchhofer) [reply keda/templates/metrics-server/apiservice.yaml:6]: The annotation `cert-manager.io/inject-ca-from` has the value `namespace-name/certificate-name`

In the current implementation of the chart the value is `namespace-name/secret-name` which seems also working but it does not follow the documentation.

Comment (JorTurFer) [reply keda/templates/metrics-server/apiservice.yaml:6]: you are right

**Comment by JorTurFer** [comment]
I believe that we will agree with a [v3 of the chart](https://github.com/kedacore/charts/issues/182). If finally we continue with the new version, I think that it'll be the good moment to address this change too

**Comment by mkilchhofer** [comment]
Rebased :-)

**Comment by JorTurFer** [comment]
I still think that we should work on KEDA chart v3 and remove all this code that we think it's not okey. It brings us the flexibility to break everything. 
I like your approach of removing the intermediate issuer, but it's a breaking change for users who are using the secret with their own CA to generate the certificates...
@tomkerkhove @zroubalik , do we have consensus about working on a chart v3 from scratch (this will be a breaking change and also will detach versioning between core and chart)

**Comment by zroubalik** [comment]
> I still think that we should work on KEDA chart v3 and remove all this code that we think it's not okey. It brings us the flexibility to break everything. I like your approach of removing the intermediate issuer, but it's a breaking change for users who are using the secret with their own CA to generate the certificates... @tomkerkhove @zroubalik , do we have consensus about working on a chart v3 from scratch (this will be a breaking change and also will detach versioning between core and chart)

I am all in for Charts simplification. I am not sure what we should do with the versioning though...mabye we can even start thinking about KEDA v3? ... there are several proposal that would be breaking and actually great https://github.com/kedacore/keda/issues/4771
But I don't know...

**Comment by mkilchhofer** [comment]
Reusing an existing issuer is now possible via:
- https://github.com/kedacore/charts/pull/530

Closing my issue now 👍

**Pull request closed without merging**